gulpfile.js

```diff
@@ -0,0 +1,157 @@
+const fs = require("fs");
+const path = require("path");
+const gulp = require("gulp");
+const ts = require("gulp-typescript");
+const sourcemaps = require("gulp-sourcemaps");
+const clean = require("gulp-clean");
+const deleteEmpty = require("delete-empty");
+const childProcess = require("child_process");
+const log = require("fancy-log");
+const clc = require("cli-color");
+const { promisify } = require("util");
+
+const sources = "packages";
+
+const documentationComponent = "documentation-component";
+const odataReact = "odata-react";
+const markdownRenderEngine = "markdown-render-engine";
+const openApiRenderEngine = "open-api-render-engine";
+const asyncApiRenderEngine = "async-api-render-engine";
+const odataRenderEngine = "odata-render-engine";
+
+const packageNames = {
+  [documentationComponent]: documentationComponent,
+  [odataReact]: odataReact,
+  [markdownRenderEngine]: `dc-${markdownRenderEngine}`,
+  [openApiRenderEngine]: `dc-${openApiRenderEngine}`,
+  [asyncApiRenderEngine]: `dc-${asyncApiRenderEngine}`,
+  [odataRenderEngine]: `dc-${odataRenderEngine}`,
+};
+const packages = {
+  [documentationComponent]: ts.createProject(
+    `${sources}/${documentationComponent}/tsconfig.json`,
+  ),
+  [odataReact]: ts.createProject(`${sources}/${odataReact}/tsconfig.json`),
+  [markdownRenderEngine]: ts.createProject(
+    `${sources}/${markdownRenderEngine}/tsconfig.json`,
+  ),
+  [openApiRenderEngine]: ts.createProject(
+    `${sources}/${openApiRenderEngine}/tsconfig.json`,
+  ),
+  [asyncApiRenderEngine]: ts.createProject(
+    `${sources}/${asyncApiRenderEngine}/tsconfig.json`,
+  ),
+  [odataRenderEngine]: ts.createProject(
+    `${sources}/${odataRenderEngine}/tsconfig.json`,
+  ),
+};
+
+const modules = Object.keys(packages);
+const distId = process.argv.indexOf("--dist");
+const dist = distId < 0 ? sources : process.argv[distId + 1];
+
+gulp.task("install:packages", async () => {
```

Comment:
why don't we just use lerna's `bootstrap?` it's more versatile, and we can set it to use `yarn` automatically, whereas here we do node's `exec`, which is to say it bluntly kinda risky

Reply:
Lerna is slower than installing packages one by one by gulp.